src/foam/core/AbstractAgency.js

```diff
@@ -36,7 +36,7 @@ foam.CLASS({
       if ( ! TASK_QUEUE.containsKey(key) ) {
         var task = new TimerTask() {
           public void run() {
-            X oldX = ((ProxyX) XLocator.get()).getX();
+            X oldX = ((MutableX) XLocator.get()).getX();
```

Comment:
Would a ProxyX actually be better in this case?

Reply:
What do users of an `XLocator` usually expect? A mutable or non-mutable X?